eZ/Publish/Core/Repository/RoleService.php

```diff
@@ -434,6 +825,32 @@ public function loadRole($id)
     }
 
     /**
+     * Loads a role draft for the given identifier.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the authenticated user is not allowed to read this role
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to read this role
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if a role with the given name was not found
+     *
+     * @param string $identifier
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function loadRoleDraftByIdentifier($identifier)
```

Comment:
> So, like this then? ..

yes something like that

> I don't see how that can work here - the DB will refuse to store on draft conflicts, and we need to store drafts when editing...?

And what is the problem? this is exactly what you are already doing in https://github.com/ezsystems/ezpublish-kernel/pull/1376/files#r38186539 or?